test/test_biogeochemistry.jl

```diff
@@ -12,6 +12,26 @@ import Oceananigans.Biogeochemistry:
     biogeochemical_advection_scheme, 
     update_biogeochemical_state!
 
+@kernel function update_PAR!(PAR, grid, P, t, bgc) 
+    i, j = @index(Global, NTuple) 
+    
+    surface_PAR = bgc.surface_PAR(t)
+    
+    z = grid.zᵃᵃᶜ[grid.Nz]
+    
+    ∫chl = - P[i, j, grid.Nz]^bgc.phytoplankton_light_attenuation_exponent*z
+    PAR[i, j, grid.Nz] =  surface_PAR*exp(bgc.water_light_attenuation_coefficient * z - bgc.phytoplankton_light_attenuation_coefficient * ∫chl)
+    for k=grid.Nz-1:-1:1
+        z = grid.zᵃᵃᶜ[k]
+        dz = grid.zᵃᵃᶜ[k+1] - z 
```

Comment:
Same as above. In general, one should _never_ reference the property `zᵃᵃᶜ`. Instead, use functions to extract nodes and metrics.